app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala

```diff
@@ -391,11 +392,15 @@ class CatalogueController @Inject() (
             )
         }
 
+      // slugs build before the inclusion of the dependency graph data wont have build deps, so they'll still need to retrieved
+      // the old api.
+      val optLegacyLatestDependencies = if(librariesOfLatestSlug.exists(_.scope.contains(DependencyScope.Build))) None else optMasterDependencies
+
       Ok(
         serviceInfoPage(
           serviceName                = serviceName,
           repositoryDetails          = repositoryDetails.copy(jenkinsURL = jenkinsLink),
-          optMasterDependencies      = optMasterDependencies,
+          optMasterDependencies      = optLegacyLatestDependencies,
```

Comment:
can we rename the parameter of serviceInfoPage to `optLegacyLatestDependencies` too? It would make it clear that it's fallback data when editing the page too.